lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -296,21 +277,11 @@ static Value lowerFullyAssociativeOp(Operation &op, OperandRange operands,
     break;
   default:
     auto firstHalf = operands.size() / 2;
-    lhs = lowerFullyAssociativeOp(op, operands.take_front(firstHalf), newOps,
-                                  operandMap, options);
-    rhs = lowerFullyAssociativeOp(op, operands.drop_front(firstHalf), newOps,
-                                  operandMap, options);
+    lhs = lowerFullyAssociativeOp(op, operands.take_front(firstHalf), newOps);
+    rhs = lowerFullyAssociativeOp(op, operands.drop_front(firstHalf), newOps);
     break;
   }
 
-  if (operandMap[lhs] + operandMap[rhs] >
-      options.maximumNumberOfVariadicOperands) {
-    if (lhs.getDefiningOp())
-      lowerUsersToTemporaryWire(*lhs.getDefiningOp(), operandMap);
-    if (rhs.getDefiningOp())
-      lowerUsersToTemporaryWire(*rhs.getDefiningOp(), operandMap);
-  }
-
   OperationState state(op.getLoc(), op.getName());
   state.addOperands(ValueRange{lhs, rhs});
```

Comment:
There was a bug here: When `lhs` and `rhs` were spilled to wires in a if-statement above, `lhs` and `rhs` refer old values.